libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSession.kt

```diff
@@ -12,17 +11,18 @@ import com.mapbox.navigator.NavigationStatus
 internal interface TripSession {
 
     val tripService: TripService
-    val locationEngine: LocationEngine
-    val locationEngineRequest: LocationEngineRequest
     var route: DirectionsRoute?
 
     fun getRawLocation(): Location?
     fun getEnhancedLocation(): Location?
     fun getRouteProgress(): RouteProgress?
     fun getState(): TripSessionState
 
-    fun start()
-    fun stop()
+    fun startActiveGuidance()
+    fun stopActiveGuidance()
+    fun startFreeDrive()
+    fun stopFreeDrive()
```

Comment:
I don't think we should make this distinction. Developers should not need to think about whether they are in free drive or active guidance from the location samples stream perspective. They are only concerned about getting the best possible location back.

In the `MapboxTripSession` docs for `startFreeDrive` you mentioned: "Start location updates without map matching.", which is not the case, we are map matching in free drive as well. `MapboxNavigaiton docs for `startFreeDrive` mention the opposite though, I think some cleanup is needed here.

If I understand correctly, that the goal was to expose a way to request only raw location updates? In my opinion, it complicates the setup, exposes an additional set of APIs that need to be sequenced correctly but doesn't provide real value because developers can just get the `LocationEngine` instance and raw request location updates themselves if they are not interested in map-matching that we're doing.

Reply:
Yeah pablo had similar feedback on the naming of these. I broke out this change into a separate pull request.

The reason I'm looking to do something like this, is because when moving LocationEngine into NavigationOptions - all the examples still need to request a location from the LocationEngine. https://github.com/mapbox/mapbox-navigation-android/pull/3248

So what would be nice, is to have a way to receive location updates before starting a session. I've broke out a proposal for this in a separate pull request https://github.com/mapbox/mapbox-navigation-android/pull/3248

> exposes an additional set of APIs that need to be sequenced correctly but doesn't provide real value because developers can just get the LocationEngine instance and request raw location updates themselves if they are not interested in map-matching that we're doing.

This is what this pull request does: https://github.com/mapbox/mapbox-navigation-android/pull/3248. Agree clients can build their own interactions with LocationEngine. But the setup is still complicated, and there are WeakReferences to prevent memory leaks.

Reply:
There's nothing preventing us from starting the service and a free-drive session right away. We don't necessarily have to wait for the route to start the trip session.

Reply:
Yeah agree, changed the function names. Because a "trip session" reflects our Active Guidance and Free Drive state.

`startLocationUpdates`, is something that can be hooked up to activity lifecycle.